Source/Charts/Utils/ViewPortHandler.swift

```diff
@@ -443,6 +454,37 @@ open class ViewPortHandler: NSObject
         return (_contentRect.origin.y + _contentRect.size.height) >= normalizedY
     }
     
+    /**
+     A method to check whether a line between two coordinates intersects with the view port.
+     
+     - Parameters:
+        - startPoint: the start coordinate of the line.
+        - endPoint: the end coordinate of the line.
+     */
+    @objc open func isIntersectingLine(from startPoint: CGPoint, to endPoint: CGPoint) -> Bool
+    {
+        // If the start or endpoint fall within the viewport, bail out early.
+        if isInBounds(point: startPoint) || isInBounds(point: endPoint) { return true }
+        
+        // Calculate the slope of the line.
+        let slope = (endPoint.y - startPoint.y) / (endPoint.x - startPoint.x)
+
+        // Check for colission with the left edge of the view port.
+        if isInBoundsY((slope * (contentRect.minX - startPoint.x)) + startPoint.y) { return true }
```

Comment:
sorry my math is a little rusty here, can you explain what is `contentRect.minX - startPoint.x` stands for here?

I'd expect details about the calculation, maybe a screenshot of the points you want to calculate could explain all

Reply:
It will calculate the `y-value` of the sloped line, by using the `x-value` of the left edge of the view port. It will then check if that t-value `isInBoundsY`. If so, the line intersects the left edge of the view port.

Reply:
hi @4np, yes, I just couldn't remember what point location is for `contentRect.minX - startPoint.x`, if we want `left edge of the view port`, isn't it `contentRect.minX`?

Reply:
In #ababe6f I updated the method that determines whether or not the line intersects the view port by:

- explicitly using the linear function `y = ax + b` and its variables `a` and `b`
- quoting the equations used
- as lines run from left to right, skip unnecessary checking for collision with the right edge of the view port

Reply:
I have added a check for `startPoint.x == endPoint.x`
looks good to merge now.